Source/Core/Core/State.cpp

```diff
@@ -323,8 +323,10 @@ static void CompressAndDumpState(CompressAndDumpState_args save_args)
 
     if (!File::Rename(filename, File::GetUserPath(D_STATESAVES_IDX) + "lastState.sav"))
       Core::DisplayMessage("Failed to move previous state to state undo backup", 1000);
-    else
+    else if (File::Exists(filename + ".dtm"))
+    {
```

Comment:
If you decide to leave those curlies in there, the `if` also needs them. But since its only a single line, you could remove them as well.